app/helpers/practices_helper.rb

```diff
@@ -148,4 +148,57 @@ def get_tool_resource_text(resource_type)
     }
     content[resource_type.to_sym]
   end
+
+  def update_category_usages
+    s_query = params["query"]
+    cat_rec = Category.where("name ILIKE ?", s_query.downcase).first
+    return if cat_rec.blank?
+    cat_id = cat_rec.id
+    last_ahoy_event = AhoyEvent.where(name: 'Category selected').last(1)
+    if last_ahoy_event.blank?
+      store_chosen_categories(s_query, nil) unless s_query.blank?
+    else
+      event_tm = last_ahoy_event[0].time
+      ev_user_id = last_ahoy_event[0].user_id
+      last_ev_cat_id = last_ahoy_event[0].properties["category_id"]
+      same_cat_id = cat_id == last_ev_cat_id
+      same_visit = (DateTime.now.to_time.utc - event_tm) <= 2
+      same_user = current_user.id == ev_user_id
+      if !same_cat_id || !same_visit || !same_user
+        store_chosen_categories(s_query, nil) unless s_query.blank?
+      end
+    end
+  end
+
+  def store_chosen_categories(s_query, chosen_categories)
+    match_search_query_to_categories s_query
+    if chosen_categories.present?
+      chosen_categories.each do |chosen_cat|
+        cat_rec = Category.find_by(name: chosen_cat)
+        ahoy.track('Category selected', {category_id: cat_rec.id }) unless cat_rec.blank?
+      end
+    end
+  end
+
+  def match_search_query_to_categories(s_query)
```

Comment:
the three methods can be refactored... I think the methods need to have one purpose. One method can do a search for a category based on the query and another method can just do the ahoy.track logic. Let me know if this makes sense.

Reply:
I refactored in to two methods for now.